http.go

```diff
@@ -0,0 +1,52 @@
+package authcontrol
+
+import (
+	"maps"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/jwtauth/v5"
+	"github.com/go-chi/traceid"
+	"github.com/go-chi/transport"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+// Service-to-service HTTP client for internal communication between Sequence services.
+func S2SClient(serviceName string, jwtSecret string, debugRequests bool) *http.Client {
```

Comment:
Maybe we could turn this into Opts struct too and add more things to it down the road